pyccel/parser/semantic.py

```diff
@@ -1114,10 +1114,10 @@ def _handle_PyccelOperator(self, expr, **settings):
     def _visit_PyccelAdd(self, expr, **settings):
         args = [self._visit(a, **settings) for a in expr.args]
         if isinstance(args[0], (TupleVariable, PythonTuple, Tuple, List)):
-            expr_new = Concatenate(args, True)
-            errors.report(PYCCEL_RESTRICTION_TODO, symbol=expr,
-                bounding_box=(self._current_fst_node.lineno, self._current_fst_node.col_offset),
-                severity='fatal', blocker=self.blocking)
+            tuple_args = []
+            for a in args:
+                tuple_args += a.get_vars() if isinstance(a, TupleVariable) else a.args
```

Comment:
Shouldn't a list comprehension be faster (and more readable) here?

Reply:
Yes, but it would require unpacking in a list comprehension which python doesn't allow

Reply:
Right! :sweat_smile: And what about this?
```
f = lambda a: a.get_vars() if isinstance(a, TupleVariable) else a.args
tuple_args = [ai for a in args for ai in f(a)]
```
Or a little more compactly (but maybe harder to read):
```
tuple_args = [ai for a in args for ai in (a.get_vars() if isinstance(a, TupleVariable) else a.args)]
```

Reply:
Won't that produce the same output?
[f(a) for a in args] will give [[a,b,c], [d,e]]
so ai in f(a) will be [a,b,c], then [d,e]

Reply:
Yes, but there is a double loop...

Reply:
Yes but the second loop doesn't unpack anything either. This effectively does:
```
a = [[1,2], [3,4]]
b = [ai for ai in a]
```
as f(a) returns a list

Reply:
Ah! Ok, I have understood what is going on now. You are right this works

Reply:
```
In [1]: args = [[1, 2, 3], [4, 5, 6]]                                                                       

In [2]: tuple_args = [ai for a in args for ai in a]                                                         

In [3]: tuple_args                                                                                          
Out[3]: [1, 2, 3, 4, 5, 6]
```